src/navigation/reroute.ts

```diff
@@ -304,13 +318,14 @@ export function reroute(
 
                   throw err;
                 }
-              );
+              )
+              .then(finishUpAndReturn);
```

Comment:
I think this might be a fix for a bug that was accidentally introduced in single-spa@6 with the profiler code. The finishUpAndReturn sets the resolved value of the promise chain, so it should be at the end.